### PR TITLE
[Feature]Support  plugin layers for backbone 

### PR DESCRIPTION
**Pull request by wanghonglie**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Support plugin layers for backbone.

## Modification

Please briefly describe what modification is made in this PR.
- Make plugins for backbone ``stage_idx`` th stage.
- Add config `yolov5_s-v61-attn0011-drop0011_syncbn_8xb16-300e_coco.py` contains DropBlock and GeneralizedAttention Block。

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

        An example of plugins format could be:

        Examples:
            >>> plugins=[
            ...     dict(cfg=dict(type='xxx', arg1='xxx'),
            ...          stages=(False, True, True, True)),
            ...     dict(cfg=dict(type='yyy'),
            ...          stages=(True, True, True, True)),
            ... ]
            >>> model = YOLOv5CSPDarknet()
            >>> stage_plugins = self.make_stage_plugins(plugins, 0, setting)
            >>> assert len(stage_plugins) == 3

        Suppose ``stage_idx=0``, the structure of blocks in the stage would be:

        .. code-block:: none

            conv1 -> conv2 -> conv3 -> yyy

        Suppose 'stage_idx=1', the structure of blocks in the stage would be:

        .. code-block:: none

            conv1 -> conv2 -> conv3 -> xxx -> yyy

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects, like MMDetection or MMClassification.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: configs/yolov5/yolov5_s-v61-attn0011-drop0011_syncbn_8xb16-300e_coco.py [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+        plugins=[
+            dict(
+                cfg=dict(type='mmdet.DropBlock', drop_prob=0.1, block_size=11),
+                stages=(False, False, True, True)),
+            dict(
+                cfg=dict(
+                    type='mmdet.GeneralizedAttention',
+                    spatial_range=-1,
+                    num_heads=8,
+                    attention_type='0010',
```
Comment (RangeKing) [inline configs/yolov5/yolov5_s-v61-attn0011-drop0011_syncbn_8xb16-300e_coco.py:55]: Need to be consistent with the config name.

File: configs/yolov5/yolov5_s-v61-attn0011-drop0011_syncbn_8xb16-300e_coco.py [hunk 2]
```diff
@@ -0,0 +1,251 @@
+_base_ = '../_base_/default_runtime.py'
```
Comment (RangeKing) [inline configs/yolov5/yolov5_s-v61-attn0011-drop0011_syncbn_8xb16-300e_coco.py:1]: It's better to inherit the `_base_ = ./yolov5_s-v61_syncbn_8xb16-300e_coco.py` config?

**Review by RangeKing — COMMENTED** [review]
The example tutorials should be modified accordingly.

**Comment by hhaAndroid** [comment]
@wanghonglie Please add UT

File: configs/yolov5/yolov5_s-v61-attn0011-drop_syncbn_8xb16-300e_coco.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
+_base_ = './yolov5_s-v61_syncbn_8xb16-300e_coco.py'
```
Comment (hhaAndroid) [inline configs/yolov5/yolov5_s-v61-attn0011-drop_syncbn_8xb16-300e_coco.py:1]: configs/yolov5/yolov5_s-v61_dropblock-stage0011_syncbn_fast_8xb16-300e_coco.py

File: docs/en/advanced_guides/how_to.md [hunk 4]
```diff
@@ -1 +1,27 @@
...
+```python
+_base_ = './yolov5_s-v61_syncbn_8xb16-300e_coco.py'
+
+model = dict(
+    backbone=dict(
+        plugins=[
+            dict(
+                cfg=dict(type='mmdet.DropBlock', drop_prob=0.1, block_size=11),
+                stages=(False, False, True, True)),
+            dict(
```
Comment (hhaAndroid) [inline docs/en/advanced_guides/how_to.md:16]: delete

File: docs/en/advanced_guides/how_to.md [hunk 5]
```diff
@@ -1 +1,56 @@
...
+                    type='mmdet.GeneralizedAttention',
+                    spatial_range=-1,
+                    num_heads=8,
+                    attention_type='0011',
+                    kv_stride=2),
+                stages=(False, False, True, True)),
+        ], ))
+```
+
+`cfg`parameter indicates the specific configuration of the plug-in. The `stages` parameter indicates whether to add plug-ins after the corresponding stage of the backbone. The length of list `stages` must be the same as the number of backbone stages.
```
Comment (PeterH0323) [inline docs/en/advanced_guides/how_to.md:24]:
```suggestion
`cfg` parameter indicates the specific configuration of the plug-in. The `stages` parameter indicates whether to add plug-ins after the corresponding stage of the backbone. The length of list `stages` must be the same as the number of backbone stages.
```

File: docs/zh_cn/advanced_guides/how_to.md [hunk 6]
```diff
@@ -1 +1,56 @@
-# how to
+本教程收集了任何如何使用 MMYOLO 进行 xxx 的答案。 如果您遇到有关`如何做`的问题及答案，请随时更新此文档！
+
+## 给骨干网络增加插件
+
+MMYOLO支持在BackBone的不同Stage后增加如none_local、dropout等插件，用户可以直接通过修改配置文件中backbone的plugins参数来实现对插件的管理。例如为`YOLOv5`增加DropBlock和GeneralizedAttention插件，其配置文件如下：
```
Comment (PeterH0323) [inline docs/zh_cn/advanced_guides/how_to.md:5]:
```suggestion
MMYOLO 支持在 BackBone 的不同 Stage 后增加如 `none_local`、`dropout` 等插件，用户可以直接通过修改配置文件中 `backbone` 的 `plugins` 参数来实现对插件的管理。例如为 YOLOv5 增加 `DropBlock` 和 `GeneralizedAttention` 插件，其配置文件如下：
```

File: docs/zh_cn/advanced_guides/how_to.md [hunk 7]
```diff
@@ -1 +1,56 @@
...
+                    type='mmdet.GeneralizedAttention',
+                    spatial_range=-1,
+                    num_heads=8,
+                    attention_type='0011',
+                    kv_stride=2),
+                stages=(False, False, True, True)),
+        ], ))
+```
+
+`cfg`参数表示插件的具体配置，`stages`参数表示是否在backbone对应的stage后面增加插件，列表`stages`的长度需要和backbone的stage数量相同。
```
Comment (PeterH0323) [inline docs/zh_cn/advanced_guides/how_to.md:24]:
```suggestion
`cfg` 参数表示插件的具体配置，`stages` 参数表示是否在 backbone 对应的stage 后面增加插件，列表 `stages` 的长度需要和 `backbone` 的 `stage` 数量相同。
```

File: docs/zh_cn/advanced_guides/how_to.md [hunk 8]
```diff
@@ -1 +1,56 @@
...
+
+## 应用多个Neck
+
+如果你想堆叠多个Neck，可以直接在配置文件中的Neck参数，MMYOLO支持以`List`形式拼接多个Neck配置，你需要保证的是上一个Neck的输出通道与下一个Neck的输入通道相匹配。如需要调整通道，可以插入`mmdet.ChannelMapper`模块用来对齐多个Neck之间的通道数量。具体配置如下：
```
Comment (PeterH0323) [inline docs/zh_cn/advanced_guides/how_to.md:28]:
```suggestion
如果你想堆叠多个 Neck，可以直接在配置文件中的 Neck 参数，MMYOLO 支持以 `List` 形式拼接多个 Neck 配置，你需要保证的是上一个 Neck 的输出通道与下一个 Neck 的输入通道相匹配。如需要调整通道，可以插入 `mmdet.ChannelMapper` 模块用来对齐多个 Neck 之间的通道数量。具体配置如下：
```

**Review by PeterH0323 — CHANGES_REQUESTED** [review]

File: docs/zh_cn/advanced_guides/how_to.md [hunk 9]
```diff
@@ -1 +1,24 @@
-# how to
+本教程收集了任何如何使用 MMYOLO 进行 xxx 的答案。 如果您遇到有关`如何做`的问题及答案，请随时更新此文档！
+
+## 给骨干网络增加插件
+
+MMYOLO支持在BackBone的不同Stage后增加如none_local、dropout等插件，用户可以直接通过修改config中backbone的plugins参数来实现对插件的管理。例如为`YOLOv5`增加DropBlock和GeneralizedAttention插件，其配置文件如下：
```
Comment (PeterH0323) [inline docs/zh_cn/advanced_guides/how_to.md:5]:
```suggestion
MMYOLO 支持在 BackBone 的不同 Stage 后增加如 `none_local`、`dropout` 等插件，用户可以直接通过修改 config 文件中 `backbone` 的 `plugins 参数来实现对插件的管理。例如为 `YOLOv5` 增加 `DropBlock` 和 `GeneralizedAttention` 插件，其配置文件如下：
```

File: docs/zh_cn/advanced_guides/how_to.md [hunk 10]
```diff
@@ -1 +1,24 @@
...
+                    type='mmdet.GeneralizedAttention',
+                    spatial_range=-1,
+                    num_heads=8,
+                    attention_type='0011',
+                    kv_stride=2),
+                stages=(False, False, True, True)),
+        ], ))
+```
+
+`cfg`参数表示插件的具体配置，`stages`参数表示是否在backbone对应的stage后面增加插件，长度需要和backbone的stage数量相同。
```
Comment (PeterH0323) [inline docs/zh_cn/advanced_guides/how_to.md:24]:
```suggestion
`cfg` 参数表示插件的具体配置，`stages` 参数表示是否在 backbone 对应的 stage 后面增加插件，长度需要和 backbone 的 stage 数量相同。
```

**Review by PeterH0323 — CHANGES_REQUESTED** [review]

File: docs/zh_cn/advanced_guides/how_to.md [hunk 11]
```diff
@@ -1 +1,56 @@
-# how to
+本教程收集了任何如何使用 MMYOLO 进行 xxx 的答案。 如果您遇到有关`如何做`的问题及答案，请随时更新此文档！
+
+## 给骨干网络增加插件
+
+MMYOLO 支持在 BackBone 的不同 Stage 后增加如 `none_local`、`dropout` 等插件，用户可以直接通过修改 config 文件中 `backbone` 的 `plugins` 参数来实现对插件的管理。例如为 `YOLOv5` 增加 `DropBlock` 和 `GeneralizedAttention` 插件，其配置文件如下：
```
Comment (hhaAndroid) [inline docs/zh_cn/advanced_guides/how_to.md:5]: dropout -> dropblock

File: docs/zh_cn/advanced_guides/how_to.md [hunk 12]
```diff
@@ -1 +1,56 @@
...
+
+```python
+_base_ = './yolov5_s-v61_syncbn_8xb16-300e_coco.py'
+
+model = dict(
+    backbone=dict(
+        plugins=[
+            dict(
+                cfg=dict(
```
Comment (hhaAndroid) [inline docs/zh_cn/advanced_guides/how_to.md:14]: 没看到 dropblock

File: docs/zh_cn/advanced_guides/how_to.md [hunk 13]
```diff
@@ -1 +1,56 @@
...
+                    num_heads=8,
+                    attention_type='0011',
+                    kv_stride=2),
+                stages=(False, False, True, True)),
+        ], ))
+```
+
+`cfg` 参数表示插件的具体配置， `stages` 参数表示是否在 backbone 对应的 stage 后面增加插件，长度需要和 backbone 的 stage 数量相同。
+
+## 应用多个Neck
```
Comment (hhaAndroid) [inline docs/zh_cn/advanced_guides/how_to.md:26]:
```suggestion
## 应用多个 Neck
```

File: docs/zh_cn/advanced_guides/how_to.md [hunk 14]
```diff
@@ -1 +1,56 @@
...
+
+如果你想堆叠多个 Neck，可以直接在配置文件中的 Neck 参数，MMYOLO支持以 `List` 形式拼接多个 Neck 配置，你需要保证的是上一个 Neck 的输出通道与下一个 Neck 的输入通道相匹配。如需要调整通道，可以插入 `mmdet.ChannelMapper` 模块用来对齐多个 Neck 之间的通道数量。具体配置如下：
```
Comment (hhaAndroid) [inline docs/zh_cn/advanced_guides/how_to.md:28]: MMYOLO 支持以

File: mmyolo/models/backbones/base_backbone.py [hunk 15]
```diff
@@ -94,6 +105,63 @@ def build_stage_layer(self, stage_idx: int, setting: list):
...
+                required if multiple same type plugins are inserted.
+            stage_idx (int): Index of stage to build
+                If stages is missing, the plugin would be applied to all
+                stages.
+            setting (list): The architecture setting of a stage layer.
+
+        Returns:
+            list[nn.Module]: Plugins for current stage
+        """
+        in_channels = make_divisible(setting[1], self.widen_factor)
```
Comment (hhaAndroid) [inline mmyolo/models/backbones/base_backbone.py:153]: This is not reasonable to write, yolox can make mistakes

**Review by hhaAndroid — APPROVED** [review]
